src/dawn/Serialization/IIRSerializer.cpp

```diff
@@ -462,7 +482,14 @@ IIRSerializer::serializeImpl(const std::shared_ptr<iir::StencilInstantiation>& i
   using namespace dawn::proto::iir;
   proto::iir::StencilInstantiation protoStencilInstantiation;
   serializeMetaData(protoStencilInstantiation, instantiation->getMetaData());
-  serializeIIR(protoStencilInstantiation, instantiation->getIIR());
+  auto& fieldNameToBCMap = instantiation->getMetaData().getFieldNameToBCMap();
+  std::set<std::string> usedBC;
+  std::transform(
+      fieldNameToBCMap.begin(), fieldNameToBCMap.end(), std::inserter(usedBC, usedBC.end()),
+      [](std::pair<std::string, std::shared_ptr<ast::BoundaryConditionDeclStmt>> const& bc) {
```

Comment:
Since it is serializing IIR, here you need `iir::BoundaryConditionDeclStmt`. `ast::*` stuff should not be used from the outside. One should use the alias classes in either `sir::` or `iir::` namespaces, depending on which type of AST you are working with.

Reply:
makes sense. done